### PR TITLE
feat: non trusted identity sign in flow

### PR DESCRIPTION
**Pull request by OlegDO**

### CODE REVIEW

File: microservices/users/src/services/identity-provider/firebase.ts [hunk 1]
```diff
@@ -101,9 +104,34 @@ class Firebase extends Abstract {
     const profile = this.getProfile(firebaseUser);
     const identityProvider = this.getIdentityProvider(firebaseUser, type);
 
+    /**
+     * Approve non-trusted identity provider
+     */
+    void (await this.approveFirebaseUserProvider(firebaseUser));
```
Comment (MatthewPattell) [inline microservices/users/src/services/identity-provider/firebase.ts]:
```suggestion
    void this.approveFirebaseUserProvider(firebaseUser);
```

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/users/src/services/identity-provider/firebase.ts (hunk 1 above)
Comment (MatthewPattell) [reply microservices/users/src/services/identity-provider/firebase.ts]: for tests: await wait(1); // to wait void

import wait from '@lomray/client-helpers/helpers/wait';

File: microservices/users/__tests__/services/identity-provider/firebase-test.ts [hunk 2]
```diff
@@ -231,4 +239,59 @@ describe('services/sign-up', () => {
       params: { isEmailVerified: undefined, isPhoneVerified: true },
     });
   });
+
+  it('should update user account: approve for non-trusted provider email', async () => {
```
Comment (MatthewPattell) [inline microservices/users/__tests__/services/identity-provider/firebase-test.ts]: rewrite

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/users/src/services/identity-provider/firebase.ts [hunk 3]
```diff
@@ -101,9 +104,40 @@ class Firebase extends Abstract {
...
+    /**
+     * If email verified by default true for trusted provider
+     */
+    if (firebaseUser.emailVerified || !this.isShouldApproveProvider) {
+      return;
+    }
+
+    console.log('called with args:', firebaseUser.uid, {
+      emailVerified: true,
+    });
```
Comment (MatthewPattell) [inline microservices/users/src/services/identity-provider/firebase.ts]: ???

File: microservices/users/src/services/identity-provider/firebase.ts [hunk 4]
```diff
@@ -101,9 +104,40 @@ class Firebase extends Abstract {
...
+  /**
+   * Approve firebase user identity
+   * NOTE: Set emailVerified to true for preventing override
+   * non-trusted providers
+   */
+  protected async approveFirebaseUserProvider(firebaseUser: UserRecord): Promise<void> {
+    const firebase = await FirebaseSdk();
+
+    console.log('this.isShouldApproveProvider', this.isShouldApproveProvider);
+    console.log('firebaseUser.emailVerified', firebaseUser.emailVerified);
```
Comment (MatthewPattell) [inline microservices/users/src/services/identity-provider/firebase.ts]: ????

**Review by MatthewPattell — CHANGES_REQUESTED** [review]

File: microservices/users/src/services/identity-provider/firebase.ts (hunk 3 above)
Comment (OlegDO) [reply microservices/users/src/services/identity-provider/firebase.ts]: I lost my eyes

**Review by MatthewPattell — APPROVED** [review]

**Pull request merged**